docs/listeners.rst

```diff
@@ -9,7 +9,11 @@ Consider that Eris performs (by default) 100 iterations for each ``forAll()`` in
 
 * ``startPropertyVerification()`` is called before the first iteration starts.
 * ``endPropertyVerification($ordinaryEvaluations)`` is called when no more iterations will be performed, both in the case of test success and failure. The ``$ordinaryEvaluations`` parameter provides the actual number of evaluations performed. This number may be less than than the number of iterations due to failures or ``when()`` filters not being satisfied.
-* ``newGeneration(array $generatedValues, $iteration)`` is called after generating a new iteration, and is passed the tuple of generated values which corresponds to the argument of ``then()`` along with the 0-based index of the iteration.
+* ``newGeneration(array $generation, $iteration)`` is called after generating a new iteration, and is passed the tuple of values along with the 0-based index of the iteration.
+* ``failure(array $generation, Exception $e)`` is called after the failure of an assertion (and not for generic exceptions). The method can be called only once per ``then()`` run.
```

Comment:
you mean the variable names? That shouldn't a relevant part of the signature afaik